### PR TITLE
WIP overhaul error stuff

### PR DESCRIPTION
**Pull request by richsalz**
Run perltidy on util/mkerr
Change some mkerr flags, write better documentation
Make generated tables "const" when genearting lib-internal ones.
Re-order creation/destruction of the hashtable, to simplify the code.

### CODE REVIEW

File: util/mkerr.pl [hunk 1]
```diff
@@ -6,325 +6,314 @@
...
+
+    -recurse    scan a preconfigured set of files for error and function codes.
+                If not specified, files on commandline are used.
+
+    -reindex    Ignore previously assigned values (except for R records in
+                the config file and renumber everything starting at 100.
+
+    -nowrite    Do not rewrite the header/source files.
+
+    -internal   Generate code that is to be built as part of OpenSSL itself.
```
Comment (levitte) [inline util/mkerr.pl]: This set of options has added indentation.  Dunno if that was intentional, but try to have them all be consistent.

File: util/mkerr.pl [hunk 2]
```diff
@@ -6,325 +6,314 @@
...
+    -debug      Verbose output debugging on stderr.
+
+    -rebuild    Force a rebuild all header and C source files, even if there
+                were no changes.
+
+    -recurse    scan a preconfigured set of files for error and function codes.
+                If not specified, files on commandline are used.
+
+    -reindex    Ignore previously assigned values (except for R records in
+                the config file and renumber everything starting at 100.
```
Comment (levitte) [inline util/mkerr.pl]: Close parenthesis

File: util/mkerr.pl [hunk 3]
```diff
@@ -338,253 +327,246 @@
...
+          " */\n",
+          "\n",
+          "#ifndef HEADER_${lib}_ERR_H\n",
+          "# define HEADER_${lib}_ERR_H\n",
+          "\n",
+          "# ifdef  __cplusplus\n",
+          "extern \"C\" {\n",
+          "# endif\n",
+          "\n",
+          "/* BEGIN ERROR CODES */\n";
```
Comment (levitte) [inline util/mkerr.pl]: Please convert this horror to a *HERE* document, i.e.:

``` perl
        push @out, <<"EOF";
/*
 * Copyright 1995-$YEAR The OpenSSL Project Authors. All Rights Reserved.
 *
 * Licensed under the OpenSSL license (the "License").  You may not use
...
EOF
```

**Comment by richsalz** [comment]
sorry, this is a WIP.  Will clean it up later when ready.

**Pull request closed without merging**